### DISCUSSION TITLE
Proposal: Online Map Editor for Endless Sky

### DISCUSSION THREAD

**Discussion by mOctave** [endless-sky/endless-sky#7215] [Ideas]
The current map editor is a really useful tool, but it can be quite hard to actually install. I installed it a couple of years ago, and the process took more than an hour because I had never worked with QT Creator before, and there were a few deprecated things that I tried to replace and, in so doing, introduced bugs.

It would be nice if there was a version of the Endless Sky map editor available online (probably written in Javascript) that could be used by anyone with a keyboard, a mouse, and a large enough screen. This is beyond my modest Javascript abilities, but I'd love it if someone else wanted to work on such a project, and I'd be happy to help where I could.

**Comment by RisingLeaf** [comment]
More an issue/feature request than a discussion

**Reply by mOctave** [reply]
Yes, except that there might be a good reason for there not to be an online map editor, and it isn't about the game specifically.

**Comment by DarcyManoel** [comment]
I should have posted this here much sooner, but nonetheless:
https://adde-endless-sky-tools.github.io/Endless-Sky-Map-Viewer/
It's not as feature rich as I would like it to be and not nearly as rich as existing editors but what it does, it does simply.

**Comment by andreadev-it** [comment]
I've been working on something like that too actually, even if it's just a viewer for now.
It's pretty much alpha, but it works. You need to open the whole endless sky code directory (or at least a directory that contains both the `data` and `images` directory) to work properly. If you only open the data folder it will show the system, but without the images.
Here's the link:
https://andreadev-it.github.io/es-data-viewer/

To do it I've also created a parser for endless sky data files in typescript. It still isn't on github, but it's up on npm:
https://www.npmjs.com/package/es-data-parser
(this is all very recent and still hasn't a good README on either repo, I'm working on it)
